src/snowflake/connector/s3_storage_client.py

```diff
@@ -92,6 +93,9 @@ def __init__(
     def transfer_accelerate_config(
         self, use_accelerate_endpoint: bool | None = None
     ) -> bool:
+        # accelerate cannot be used in China and us government
+        if "S3China" in self.location_type:
```

Comment:
nit: stage_info.get returns None if locationType is not in the dict, so we shall check whether self.location_type has a value
```suggestion
        if self.location_type and "S3China" in self.location_type:
```